### PR TITLE
Update to tensorflow 1.3

### PR DESCRIPTION
**Pull request by fkm3**
* Tested on linux without Docker.
* Couldn't get nix build to work, so I just updated the URL and hash.
* Did not test macos build.

The mnist change was necessary because the argmax output type is now polmorphic.

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # ChangeLog
 
 ## Upcoming (v0.2.0.0)
+- Support to tensorflow 1.3.
```
Comment (blackgnezdo) [inline ChangeLog.md]: Switch to?

File: tensorflow-core-ops/Setup.hs [hunk 2]
```diff
@@ -90,7 +90,14 @@ fudgePackageDesc lbi p = p
 
 blackList =
     [ -- Requires the "func" type:
```
Comment (blackgnezdo) [inline tensorflow-core-ops/Setup.hs]: Are all of the below in this category?

File: tensorflow-mnist/app/Main.hs [hunk 3]
```diff
@@ -75,7 +76,7 @@ createModel = do
     logitBiases <- TF.zeroInitializedVariable [numLabels]
     let logits = (hidden `TF.matMul` TF.readValue logitWeights)
                  `TF.add` TF.readValue logitBiases
-    predict <- TF.render $ TF.cast $
+    predict <- TF.render @TF.Build @LabelType $
```
Comment (blackgnezdo) [inline tensorflow-mnist/app/Main.hs]: Is this a drive-by nicety or is it somehow required for 1.3 transition?

Comment (blackgnezdo) [reply tensorflow-mnist/app/Main.hs]: Ah, re-reading the description helped :)

File: ChangeLog.md (hunk 1 above)
Comment (fkm3) [reply ChangeLog.md]: Done.

File: tensorflow-core-ops/Setup.hs (hunk 2 above)
Comment (fkm3) [reply tensorflow-core-ops/Setup.hs]: Yeah, Judah plugged all of the other holes apparently :)
